docs/contributing/mentorship.md

```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
+5. Once you receive replies from most of them, review again with your co-mentor and shortlist candidates that you would now like to interview.
+6. Conduct 20-minute interviews (refer to [recommendations](#recommendations) for what to keep in mind during these interviews) of the shortlisted mentees to get to know them even better, and finally select a single person
+7. Send an email to the selected mentee asking for confirmation, and if they are happy with this.
+8. Once they reply, fill out the selection form in the relevant portal! :)
+
+#### Recommendations
+
+During mentee selection for our various mentorship programs, we want to follow some best practices to ensure that the mentorship is a fruitful experience for both mentees and mentors. A large part of ensuring this is selecting the most suitable candidate for the mentorship!
+
+This process is hard and requires time, as it involves holistically looking at mentorship applications, and trying to ascertain if they would be able to benefit from this mentorship and ideally become a great future contributor or even maintainer.
+
+With several past mentorships, we have been able to distill some learnings from the process, which are listed here as recommendations to mentors,
+
+* Encourage mentees who are interested in applying, by asking them to look through our issues and trying to work on one in which they would learn a bit about the codebase!
+* Allocate some time upfront for this process, as you might have a lot of mentee applications to look through.
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
```

Comment:
This is neither required nor expected. We can note that having prior contributions is favorable because it demonstrates willingness to engage with the community and independence.

Reply:
Let me rephrase here.